documentation/dataplane/dash-flow/DASH_FLOW_SAI.md

```diff
@@ -0,0 +1,615 @@
+    SAI_FLOW_METADATA_ATTR_DEST_PA,
+
+    /* ID for metering class. Type: [sai_uint64_t] */
+    SAI_FLOW_METADATA_ATTR_METERING_CLASS,
+
+    /* Information required for rewriting. Type: [sai_dash_rewrite_info_t] */
+    SAI_FLOW_METADATA_ATTR_REWRITE_INFO,
+
+    /* Vendor-specific metadata details. Type: [sai_u8_list_t] */
+    SAI_FLOW_METADATA_ATTR_VENDOR_METADATA,
+
+    /* Ending marker for attributes */
+    SAI_FLOW_METADATA_ATTR_END
+
+} sai_flow_metadata_attr_t;
+
+```
+
+
+
+### Flow state protobuf definiation
+
+```protobuf
+syntax = "proto3";
+
+message SaiDashFlowMetadata {
+    uint32 version = 1;
+    SaiDashPolicyResult policy_result = 2;
+    /* Destination PA IP address */
+    string dest_pa = 3; 
+    uint64 metering_class = 4;
+    uint64 metering_class2 = 5;
+    SaiDashHaRewriteInfo rewrite_info = 6;
+    /* Vendor specific metadata */
+    bytes vendor_metadata = 7; 
+}
+
+enum SaiDashPolicyResult {
+    SAI_DASH_HA_POLICY_RESULT_NONE = 0;
+    SAI_DASH_HA_POLICY_RESULT_ALLOW = 1;
+    SAI_DASH_HA_POLICY_RESULT_DENY = 2;
+}
+
+enum SaiDashHaRewriteFlags {
+    SAI_DASH_HA_REWRITE_NONE = 0; /* Default, unused value */
+    SAI_DASH_HA_REWRITE_IFLOW_DMAC = 1;
+    SAI_DASH_HA_REWRITE_IFLOW_SIP = 2;
+    SAI_DASH_HA_REWRITE_IFLOW_SPORT = 4;
+    SAI_DASH_HA_REWRITE_IFLOW_VNI = 8;
+    SAI_DASH_HA_REWRITE_RFLOW_SIP = 16;
+    SAI_DASH_HA_REWRITE_RFLOW_DIP = 32;
+    SAI_DASH_HA_REWRITE_RFLOW_DPORT = 64;
+    SAI_DASH_HA_REWRITE_RFLOW_SPORT = 128;
+    SAI_DASH_HA_REWRITE_RFLOW_VNI = 256;
+}
+
+message SaiDashHaRewriteInfo {
+    /* Bitmap of SaiDashHaRewriteFlags */
+    uint64 rewrite_flags = 1; 
+    /* Initiator Flow DMAC */
+    string iflow_dmac = 2; 
+    /* Initiator Flow Source IP address */
+    string iflow_sip = 3; 
+    /* Initiator Flow L4 Source Port */
+    uint32 iflow_sport = 4; 
+    /* Initiator Flow VNID */
+    uint32 iflow_vni = 5; 
+    /* Reverse Flow Source IP address */
+    string rflow_sip = 6; 
+    /* Reverse Flow Destination IP address */
+    string rflow_dip = 7; 
+    /* Reverse Flow Destination Port */
+    uint32 rflow_dport = 8;
+    /* Reverse Flow Source Port */
+    uint32 rflow_sport = 9; 
+    /* Reverse Flow VNID */
+    uint32 rflow_vni = 10; 
+}
+```
+
+
+
+### Flow state query type
+
+```c++
+typedef enum _sai_flow_state_query_type_t{
+    SAI_FLOW_TABLE_ENTRY_QUERY_SRCIP,
+    SAI_FLOW_TABLE_ENTRY_QUERY_SRCPORT,
+    SAI_FLOW_TABLE_ENTRY_QUERY_DSTIP,
+    SAI_FLOW_TABLE_ENTRY_QUERY_DSTPORT,
+    SAI_FLOW_TABLE_ENTRY_QUERY_PROTO,
+    SAI_FLOW_TABLE_ENTRY_QUERY_ALL,
+    SAI_FLOW_TABLE_ENTRY_QUERY_ALL_AGED,
+    SAI_FLOW_TABLE_ENTRY_QUERY_VERSION_LESS_THAN,
+    SAI_FLOW_TABLE_ENTRY_QUERY_VERSION_LESS_THAN_OR_EQUAL_TO,
+    SAI_FLOW_TABLE_ENTRY_QUERY_VERSION_EQUAL_TO,
+    SAI_FLOW_TABLE_ENTRY_QUERY_VERSION_GREATER_THAN,
+    SAI_FLOW_TABLE_ENTRY_QUERY_VERSION_GREATER_THAN_OR_EQUAL_TO
+} sai_flow_state_query_type_t;
+```
+
+
+
+## Usage of DASH Flow SAI APIs 
```

Comment:
this section could be removed, as the API's comment will have these covered already.